### PR TITLE
Allow for addons to prompt for data input

### PR DESCRIPTION
**Pull request by stevesloka**
This allows for an addon to prompt for data. My use case is registry-creds needs additional secrets to enable pulling credentials from a registry. This should vastly simplify this need by prompting the user. 

Looking for feedback on the implementation as well as I need to update the registry-creds repo to handle correctly before merging this in. 

Here's a screenshot of what it looks like:

screen shot 2017-02-17 at 1 53 12 pm

// cc #901 upmc-enterprises/registry-creds#34 @aaron-prindle

### CODE REVIEW

**Comment by aaron-prindle** [comment]
@minikube-bot ok to test

File: cmd/minikube/cmd/config/util.go [hunk 1]
```diff
@@ -88,11 +88,67 @@ func GetClientType() machine.ClientType {
...
+			negResponses := []string{"no", "n"}
+
+			enableAWSECR := AskForYesNoConfirmation("\nDo you want to enable AWS Elastic Container Registry?", posResponses, negResponses)
+			if enableAWSECR {
+				awsAccessID := AskForStaticValue("-- Enter AWS Access Key ID: ")
+				awsAccessKey := AskForStaticValue("-- Enter AWS Secret Access Key: ")
+				awsRegion := AskForStaticValue("-- Enter AWS Region: ")
+				awsAccount := AskForStaticValue("-- Enter 12 digit AWS Account ID: ")
+
+				cluster.CreateSecret(
```
Comment (aaron-prindle) [inline cmd/minikube/cmd/config/util.go]: I believe that for GCR/ECR to work in registry-creds current state, both secrets will need to be created for the pod to start properly.  This was the case in my experience at least.  Maybe empty values can be used for secrets specify which mode is enabled? We can also make changes to registry-creds itself.

Comment (stevesloka) [inline cmd/minikube/cmd/config/util.go]: I'm working on some changes to make this easier, will get back to you when I do. I just finished cleaning up the k8s interfaces and implemented the client-go. Will get back soon!

**Comment by stevesloka** [comment]
Hey @aaron-prindle, I did a bunch of rework on the registry-creds addon. We can now run both registries at the same time. Each cloud gets their own secret. 

I can't get GCR to work for me, the credentials don't work. But they also don't match the path you have in the readme. Not sure if my setup is wrong, or something else.

Could you give it a shot? I think the only remaining item I need to add to registry-creds is to watch the secrets in kube-system. What I've found is you enable the addon, then create the secrets, but the addon won't have the correct secrets until the pod is restarted. So I want to make that experience a bit better.

**Comment by aaron-prindle** [comment]
GCR is working for me on registry-creds:1.6 after I copied my ~/.config/gcloud/application_default_credentials.json files into the secret.yaml.  I did have to run `gcloud auth application-default login` to get an updated application_default_credentials.json file though in order to pull my GCR image.  I definitely agree the ability to watch for a secret would be a great addition instead of having to create the secret prior.

**Comment by aaron-prindle** [comment]
Currently I cannot build this branch though as there seems to be a build issue with client-go:
https://travis-ci.org/kubernetes/minikube/builds/205196451#L332

**Comment by aaron-prindle** [comment]
This is working for me.  It seems there is a boilerplate license issue though:
I believe that the text needs to match exactly the boilerplate of the other files including the newline format:
https://github.com/kubernetes/minikube/blob/master/hack/boilerplate/boilerplate.go.txt
Probably easiest to just copy it directly from another file and keep the newlines.

**Comment by stevesloka** [comment]
@aaron-prindle yeah somehow the spacing was off, just fixed and pushed up now.

**Comment by aaron-prindle** [comment]
LGTM

**Pull request merged**